src/main/scala/WikiTextHelper.scala

```diff
@@ -84,10 +110,14 @@ object WikiTextHelper {
       return Array((text, false))
     }
     val textParts = text.split(Pattern.quote(found.get))
-    val middle = found.get
-    val head = textParts.headOption.getOrElse("")
-    val tail = if (textParts.size == 2) textParts(1) else ""
-    splitEntity(head, entities) ++ Array((middle, true)) ++ splitEntity(tail, entities)
```

Comment:
**BUG**
This only able to extract a single name from the sentence, it doesn't work if the sentence has more than one entity.

Fix at below. Just loop through the parts instead of manually joining head, middle and tail.

Reply:
Actually it worked for multple entities, what didn't work was if the first word is an entity.